testing/scenario_app/lib/src/send_text_focus_semantics.dart

```diff
@@ -0,0 +1,104 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:typed_data';
+import 'dart:ui';
+
+import 'package:scenario_app/src/channel_util.dart';
+
+import 'scenario.dart';
+
+/// A scenario that sends back messages when touches are received.
+class SendTextFocusScemantics extends Scenario {
+  /// Constructor for `SendTextFocusScemantics`.
+  SendTextFocusScemantics(Window window) : super(window);
+
+  @override
+  void onBeginFrame(Duration duration) {
+    // Doesn't matter what we draw. Just paint white.
+    final SceneBuilder builder = SceneBuilder();
+    final PictureRecorder recorder = PictureRecorder();
+    final Canvas canvas = Canvas(recorder);
+
+    canvas.drawRect(
+      Rect.fromLTWH(0, 0, window.physicalSize.width, window.physicalSize.height),
+      Paint()..color = const Color.fromARGB(255, 255, 255, 255),
+    );
+    final Picture picture = recorder.endRecording();
+
+    builder.addPicture(
+      Offset.zero,
+      picture,
+    );
+    final Scene scene = builder.build();
+    window.render(scene);
+    scene.dispose();
+
+    // On the first frame, also pretend that it drew a text field. Send the
+    // corresponding semantics tree comprised of 1 node for the text field.
+    window.updateSemantics((SemanticsUpdateBuilder()
+      ..updateNode(
+        id: 0,
+        // SemanticsFlag.isTextField.
+        flags: 16,
+        // SemanticsAction.tap.
+        actions: 1,
+        rect: const Rect.fromLTRB(0.0, 0.0, 414.0, 48.0),
+        label: 'flutter textfield',
+        textDirection: TextDirection.ltr,
+        textSelectionBase: -1,
+        textSelectionExtent: -1,
+        platformViewId: -1,
+        maxValueLength: -1,
+        currentValueLength: 0,
+        scrollChildren: 0,
+        scrollIndex: 0,
+        transform: Float64List.fromList(<double>[1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 366.5, 0.0, 1.0]),
```

Comment:
Isn't there a function somewhere where you can make a translation matrix?  For readabilities sake you should at least have a local one.

Reply:
I just copy pasted it from a real run but it doesn't actually matter for the purpose of this test. Removed.